### ISSUE TITLE
UX: Deleting bootenvs / templates fails w/o error when referenced elsewhere

### ISSUE THREAD

**Issue by wdennis** [digitalrebar/provision#462]
Tried to delete bootenvs and templates in my DRP 3.1 system, but after selecting the items, clicking the "Delete" button and confirming, the screen returned with the items still present. As it turns out, if the items to be deleted are referenced somewhere else, they cannot be deleted until unreferenced (which is a good thing.) However, there is not error passed up into the UI about the reason the items selected could not be deleted. An informative error with detail about the references should be displayed to the user.

**Comment by sygibson** [comment]
backend implemented better error handling - which should bubble up to the UX with appropriate codes

needs validation that UX is passing the correct errors up in to UX

**Comment by galthaus** [comment]
I did the following:
1. Create a new profile aa - can be empty and no parameters.
2. Create a machine - doesn't have to be real.
3. Add profile to machine.
4. Go back to profile screen. 
5. Delete profile aa.

I found the at the profile is still in the list and not deleted.  (this is correct, but it would be nice to get feed back).
I checked the network log and the call returns 422 validation error with an error explaining the proper reason why the profile was not deleted.
